### PR TITLE
Added new topics.

### PR DESCRIPTION
**Pull request by kempb**

### CODE REVIEW

File: docs/essentials/exceptions/exceptions-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+  C#
+  ```
+  if (conn.State != ConnectionState.Closed)
+  {
+      conn.Close();
+  }
+  ```
+
+  C++
```
Comment (mairaw) [inline docs/essentials/exceptions/exceptions-best-practices.md]: I don't think we should show examples with languages that are not supported with .NET Core like C++ or VB (for now). We also should make sure that when we're porting the content, the samples we're showing use classes that are supported in .NET Core IMO.

File: docs/essentials/exceptions/exceptions-best-practices.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+- Use **try**/**catch** blocks around code that can potentially generate an exception, and use a **finally** block to clean up resources, if necessary. This way, the **try** statement generates the exception, the **catch** statement handles the exception, and the finally statement closes or deallocates resources whether or not an exception occurs.
+
+- In **catch** blocks, always order exceptions from the most specific to the least specific. This technique handles the specific exception before it is passed to a more general **catch** block.
+
+## Creating and raising exceptions
+
+The following list contains guidelines for creating your own exceptions and when they should be raised. For a more details, see [Design Guidelines for Exceptions](https://msdn.microsoft.com/library/ms229014).
+
+- Design classes so that an exception is never thrown in normal use. For example, a [FileStream](https://msdn.microsoft.com/library/system.io.filestream) class provides methods that help determine whether the end of the file has been reached. This avoids the exception that is thrown if you read past the end of the file. The following example shows how to read to the end of the file.
```
Comment (mairaw) [inline docs/essentials/exceptions/exceptions-best-practices.md]: We should link to the classes in the .NET Core API reference in these topics such as http://dotnet.github.io/api/System.IO.FileStream.html

File: docs/essentials/exceptions/exceptions-com-interop.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Handling COM interop exceptions
```
Comment (mairaw) [inline docs/essentials/exceptions/exceptions-com-interop.md]: @blackdwarf does this topic applies to .NET Core?

Comment (blackdwarf) [inline docs/essentials/exceptions/exceptions-com-interop.md]: No. COM interop doesn't exist in .NET Core.

Comment (mairaw) [inline docs/essentials/exceptions/exceptions-com-interop.md]: That's what I thought! @kempb, you can delete this one :smile:

**Comment by mairaw** [comment]
Fixes #222

**Pull request merged**